exec-service-wasmer/src/wasmer_opcode_control.rs

```diff
@@ -0,0 +1,210 @@
+    fn feed<'b>(
+        &mut self,
+        operator: Operator<'b>,
+        state: &mut MiddlewareReaderState<'b>,
+    ) -> Result<(), MiddlewareError> {
+        match operator {
+            Operator::MemoryGrow { .. } => {
+                // Check if memory limit (memory_grow >= max_memory_grow)
+                state.extend(&[
+                    Operator::GlobalGet {
+                        global_index: self.global_indexes.memory_grow().as_u32(),
+                    },
+                    Operator::I64Const {
+                        value: self.max_memory_grow as i64,
+                    },
+                    Operator::I64GeU,
+                ]);
+
+                // Insert breakpoint BREAKPOINT_VALUE_MEMORY_LIMIT if memory_grow >= max_memory_grow
+                state.extend(
+                    self.breakpoint
+                        .as_ref()
+                        .insert_breakpoint(BREAKPOINT_VALUE_MEMORY_LIMIT)
+                        .iter(),
+                );
+
+                // Increment the memory_grow counter
+                state.extend(&[
+                    Operator::GlobalGet {
+                        global_index: self.global_indexes.memory_grow().as_u32(),
+                    },
+                    Operator::I64Const { value: 1 as i64 },
+                    Operator::I64Add,
+                    Operator::GlobalSet {
+                        global_index: self.global_indexes.memory_grow().as_u32(),
+                    },
+                ]);
+
+                // Cache the memory_grow counter
+                state.extend(&[Operator::GlobalSet {
+                    global_index: self.global_indexes.memory_grow_cached().as_u32(),
+                }]);
+
+                // Check if memory limit (memory_grow_cached > max_memory_grow_delta)
+                state.extend(&[
+                    Operator::GlobalGet {
+                        global_index: self.global_indexes.memory_grow_cached().as_u32(),
+                    },
+                    Operator::I64Const {
+                        value: self.max_memory_grow_delta as i64,
+                    },
+                    Operator::I64GtU,
+                ]);
+
+                // Insert breakpoint BREAKPOINT_VALUE_MEMORY_LIMIT if memory_grow_cached > max_memory_grow_delta
+                state.extend(
+                    self.breakpoint
+                        .as_ref()
+                        .insert_breakpoint(BREAKPOINT_VALUE_MEMORY_LIMIT)
+                        .iter(),
+                );
+
+                // Retrieve cached memory_grow counter
+                state.extend(&[Operator::GlobalGet {
+                    global_index: self.global_indexes.memory_grow_cached().as_u32(),
+                }]);
+            }
+            _ => {}
+        }
+
+        state.push_operator(operator);
+
+        Ok(())
+    }
+}
+
+#[allow(dead_code)]
+pub(crate) fn reset_memory_grow(instance: &Instance) {
```

Comment:
Consider renaming to `reset_memory_grow_counter()`.

Reply:
done